File: app/src/main/java/moe/notify/animenotifier/domain/model/anime/AnimeStudio.java
package moe.notify.animenotifier.domain.model.anime;


public class AnimeStudio {

    public String name;
    public String wiki;
    public int mainStudio;
    public long id;


    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AnimeStudio)) {
            return false;
        }

        AnimeStudio that = (AnimeStudio) o;

        return (id == that.id) && ((name != null) ? name.equals(that.name) : ((that.name == null) && ((wiki != null) ? wiki.equals(that.wiki) : (that.wiki == null))));

    }

    @Override
    public int hashCode() {
        int result = (name != null) ? name.hashCode() : 0;
        result = (31 * result) + ((wiki != null) ? wiki.hashCode() : 0);
        result = (31 * result) + (int) id;
        return result;
    }
}

File: app/src/main/java/moe/notify/animenotifier/domain/model/anime/AnimeStudio.kt
package moe.notify.animenotifier.domain.model.anime

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey


open class AnimeStudio(
    open var name: String? = null,
    open var wiki: String? = null,
    open var mainStudio: Int = 0,
    @PrimaryKey
    open var id: Long = 0) : RealmObject()